### PR TITLE
changed p % E to p % last_E for collision and analog tallies

### PR DESCRIPTION
**Pull request by samuelshaner**
This PR addresses issue #497, namely changing the energy used to query the neutron velocity, `yield_delayed`, and `nu_delayed` for analog and collision tallies from `p % E` to `p % last_E`. Additionally, the units of velocity have been changed from m/s to cm/s. This is done because the velocity needed in the time-dependent neutron transport equation must have the same length units as the macroscopic cross sections. 

@nelsonag since you pointed out issue #497, I've assigned this to you.

### CODE REVIEW

File: src/tally.F90 [hunk 1]
```diff
@@ -139,12 +148,16 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
           else
             score = p % last_wgt
           end if
+
+          ! Score the flux weighted inverse velocity with velocity in units of
+          ! cm/s
           score = score / material_xs % total &
-               / (sqrt(TWO * p % E / (MASS_NEUTRON_MEV)) * C_LIGHT)
+               / (sqrt(TWO * E / (MASS_NEUTRON_MEV)) * C_LIGHT * 100.0)
```
Comment (paulromano) [inline src/tally.F90]: Use `100.0_8` to be safe

File: src/tally.F90 [hunk 2]
```diff
@@ -139,12 +148,16 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
...
 
         else
-          ! For inverse velocity, we need no cross section
-          score = flux / (sqrt(TWO * p % E / (MASS_NEUTRON_MEV)) * C_LIGHT)
+          ! For inverse velocity, we don't need a cross section. The velocity is
+          ! in units of cm/s.
+          score = flux / (sqrt(TWO * E / (MASS_NEUTRON_MEV)) * C_LIGHT * 100.0)
```
Comment (paulromano) [inline src/tally.F90]: Same thing here

**Comment by paulromano** [comment]
If you expect answers to change (seems like they should from this fix), then test results need to be updated.

**Comment by samuelshaner** [comment]
@paulromano I just push a commit that fixed the tests and addressed your comments.

**Comment by nelsonag** [comment]
Looks good to me. Thanks!

**Pull request merged**